File: 1855.maximum-distance-between-a-pair-of-values/solution.go
func maxDistance(nums1 []int, nums2 []int) int {
    i, j, res := 0, 0, 0
    for i < len(nums1) && j < len(nums2) {
        if nums1[i] <= nums2[j] {
            res = max(res, j - i)
            j++
        } else {
            i++
        }
    }
    return res
}

File: 1855.maximum-distance-between-a-pair-of-values/solution.kt
class Solution {
    fun maxDistance(nums1: IntArray, nums2: IntArray): Int {
        var res = 0
        var i = 0
        var j = 0

        while (i < nums1.size && j < nums2.size) {
            if (nums1[i] <= nums2[j]) {
                res = maxOf(res, j - i)
                j++
            } else {
                i++
            }
        }

        return res
    }
}